addons/account_sequence/models/account_move.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    name = fields.Char(compute='_compute_name')
+    highest_name = fields.Char(compute='_compute_highest_name')
+
+    def _get_journal_sequence(self):
+        if self.move_type in ('out_refund', 'in_refund') and self.journal_id.type in ('sale', 'purchase') \
+                and self.journal_id.refund_sequence_id:
+            return self.journal_id.refund_sequence_id._get_current_sequence()
+        else:
+            return self.journal_id.sequence_id._get_current_sequence()
+
+    @api.depends('state', 'journal_id', 'date')
+    def _compute_name(self):
+        # OVERRIDE
+        moves_with_ir_sequence = self.filtered('journal_id.sequence_id')
+        for move in moves_with_ir_sequence:
+            if not move.posted_before and not move.state == 'posted':
+                seq = move._get_journal_sequence()
+                move.name = seq.get_next_char(seq.number_next_actual)
+        super(AccountMove, self - moves_with_ir_sequence)._compute_name()
+
+    def _post(self, soft=True):
+        # OVERRIDE
+        moves_with_ir_sequence = self.filtered('journal_id.sequence_id')
+        for move in moves_with_ir_sequence:
+            seq = move._get_journal_sequence()
+            next_name = seq.get_next_char(seq.number_next_actual)
+            if not move.posted_before and (move.name == '/' or move.name == next_name):
+                # We only compute new name if user didn't manually change the name
```

Comment:
Not sure I understand how one can edit manually the number when using ir.sequence
It should even be readonly in the view I think

Reply:
But then it will give you the same number twice?